ouroboros-consensus-cardano-test/src/Test/Consensus/Cardano/Generators.hs

```diff
@@ -406,6 +407,14 @@ instance c ~ MockCryptoCompatByron
       injAnytimeMary    (Some      query)  = SomeSecond (QueryAnytimeMary      query)
       injHardFork       (Some      query)  = SomeSecond (QueryHardFork         query)
 
+instance Arbitrary (WithVersion (HardForkNodeToClientVersion (CardanoEras c))
```

Comment:
Can we have one `Arbitrary (WithVersion blk (SomeSecond BlockQuery blk)) => Arbitrary (WithVersion blk (SomeSecond Query blk))` instance? I anticipate it belongs somewhere in `ouroboros-consensus-test`, probably next to the `Arbitrary (SomeSecond BlockQuery blk)=> Arbitrary (SomeSecond Query blk)` instance.

Reply:
Hmmm. I find overlapping instances tend to add more confusion than they're worth. In general, I prefer explicitness for the sake of transparency.

In this case, I'm hesitant to make the existing overlapping instance _more_ general. I'm guessing the earlier logic was that there are only two `ShelleyNodeToClientVersions` and the difference between them was the addition of one query. So most of the instances were trivial, hence the overlapping was used for that.

But with the new more general instance, this would apply to _any_ version type, even if it were rich. That's not a risk right now, since we already have instances for all the other types (since there was no such general instance), but this opens the door for us to accidentally rely on this general instance for a future version type.

However, that said, this is for `Arbitrary`. So the greatest risk is that we'd see test failures for mismatched versions and values. Once we debug the failure to see the cause, we can add the corresponding override instance. So I'm optimistic here. Let's go with what you have here.